### PR TITLE
Add SSH support for units and machines

### LINKED ISSUE #79: Implement ssh
For purposes like Matrix's reboot and kill_juju_agent actions, "ssh" is probably less likely to error than "run".

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**
Fixes #79

### CODE REVIEW

File: tests/integration/test_unit.py [hunk 1]
```diff
@@ -99,6 +99,33 @@
...
 
 
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_ssh(event_loop):
+    # ensure that asyncio.subprocess will work;
+    try:
+        asyncio.get_child_watcher().attach_loop(event_loop)
+    except RuntimeError:
+        pytest.skip('test_scp will always fail outside of MainThread')
```
Comment (SimonRichardson) [inline tests/integration/test_unit.py]:
```suggestion
        pytest.skip('test_ssh will always fail outside of MainThread')
```

**Comment by johnsca** [comment]
[`provisioner.py`](https://github.com/juju/python-libjuju/blob/master/juju/provisioner.py) already contains some SSH logic using the Paramiko library.  I'm not entirely sure why that library was used instead of using an async subprocess like this PR does.  I wonder if that dependency could be removed by refactoring that code to use this approach?

**Comment by ChrisMacNaughton** [comment]
@SimonRichardson `scp` already does shell out?

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
@ChrisMacNaughton can you rebase to the latest master, our merge bot won't let this land.

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by hpidcock** [comment]
#422 fixes unit ssh async

**Pull request merged**